contracts/easy-poll/src/lib.rs

```diff
@@ -0,0 +1,1047 @@
+     **********/
+
+    /// User can update the poll if starts_at > now
+    /// it panics if
+    /// - user tries to create an invalid poll
+    /// - if poll aready exists and starts_at < now
+    pub fn create_poll(
+        &mut self,
+        iah_only: bool,
+        questions: Vec<Question>,
+        starts_at: u64,
+        ends_at: u64,
+        title: String,
+        tags: Vec<String>,
+        description: Option<String>,
+        link: Option<String>,
+    ) -> PollId {
+        let created_at = env::block_timestamp_ms();
+        require!(
+            created_at < starts_at,
+            format!("poll start must be in the future")
+        );
+        let poll_id = self.next_poll_id;
+        self.next_poll_id += 1;
+        self.initalize_results(poll_id, &questions);
+        self.polls.insert(
+            &poll_id,
+            &Poll {
+                iah_only,
+                questions,
+                starts_at,
+                ends_at,
+                title,
+                tags,
+                description,
+                link,
+                created_at,
+            },
+        );
+        poll_id
+    }
+
+    /// user can change his answer when the poll is still active
+    // TODO: currently we do not allow users to change the answer
+    /// it panics if
+    /// - poll not found
+    /// - poll not active
+    /// - poll.verified_humans_only is true, and user is not verified on IAH
+    // - user tries to vote with an invalid answer to a question
+    #[payable]
+    #[handle_result]
+    pub fn respond(
+        &mut self,
+        poll_id: PollId,
+        answers: Vec<Option<Answer>>,
+    ) -> Result<(), PollError> {
+        require!(
+            env::attached_deposit() >= RESPOND_COST,
+            "attached_deposit not sufficient"
+        );
+        let caller = env::predecessor_account_id();
+
+        match self.assert_active(poll_id) {
+            Err(err) => return Err(err),
+            Ok(_) => (),
+        };
+
+        // TODO: I think we should add a option for the poll creator to choose whether changing
+        // the answers while the poll is active is allowed or not
+        self.assert_answered(poll_id, &caller);
+        let poll = match self.polls.get(&poll_id) {
+            None => return Err(PollError::NotFound),
+            Some(poll) => poll,
+        };
+        // if iah calls the registry to verify the iah sbt
+        if poll.iah_only {
+            ext_registry::ext(self.registry.clone())
+                .is_human(caller.clone())
+                .then(
+                    Self::ext(env::current_account_id())
+                        .with_static_gas(RESPOND_CALLBACK_GAS)
+                        .on_human_verifed(true, caller, poll_id, answers),
+                );
+        } else {
+            Self::ext(env::current_account_id())
+                .with_static_gas(RESPOND_CALLBACK_GAS)
+                .on_human_verifed(false, caller, poll_id, answers);
```

Comment:
If I add an empty list to it I get an error from the environment. I assume this will just be empty if there is no call before it. I will test it in the integration tests

Reply:
Done